.github/workflows/android.yml

```diff
@@ -38,7 +38,7 @@ jobs:
           echo "Emulator starting in background"
 
       - name: Configure
-        run: cmake -Werror=dev -DCMAKE_TOOLCHAIN_FILE=$ANDROID_NDK_LATEST_HOME/build/cmake/android.toolchain.cmake -DANDROID_PLATFORM=16 -DANDROID_ABI=x86_64 -DCMAKE_BUILD_TYPE=Debug ..
```

Comment:
[actions/runner-images](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md) claims that cmake version 3.22.1 ships with Android on macos-14, but I couldn't find it. I would prefer using the builtin one (and keep the -Werror=dev flag), but at the end of the day we control neither the image nor the cmake file so removing the dev flag is fine.